### ISSUE TITLE
[feature] Reserve critical free disk space to avoid server crash

### ISSUE THREAD

**Issue by tobykurien** [nextcloud/server#47123]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
When users of a multi-user nextcloud instance upload more files than there is space for on the server, the server ends up with 0 bytes free. This causes mysql to die as it runs out of space, and the server to crash with "Internal server error". This is hard to recover from, because clients auto-upload and retry, so even if I free up some free space on the server, it will get used up immediately and the server will crash again. What is really needed is for nextcloud to reserve some critial free space so that the server doesn't crash.

**Describe the solution you'd like**
A simple feature to add would be to have a default amount of reserved free space that the nextcloud server will not use, similar to the OWNCLOUD_CRITICAL_FREE_SPACE_BYTES env variable for the client. A good default might be 250Mb, which could be configurable in config.php.

**Describe alternatives you've considered**
Adding a user quota might help for a single-user system but is not practical for multiple users where different users have different space requirements. I previously used `instance_quota` app and manually set some limit, but this app is no longer available. The only solution left to me is file system level quota like ZFS datasets, but this is painful and shouldn't be needed if this simple feature was available.

**Additional context**

**Comment by SystemKeeper** [comment]
Just to mention it, the disk can also be filled up by other services (e.g. logfiles not correctly rotated, temporary space for a backup, etc.). So even if Nextcloud reserves some space, it doesn’t mean the problem will not occur.

**Comment by tobykurien** [comment]
Agreed, however those scenarios are easy to recover from (delete log
files, etc.). In the case of Nextcloud, when you have multiple users
with photo auto-upload enabled, the server will just keep crashing.

**Comment by susnux** [comment]
In the meantime I would recommend to use a separate partition for the data, so it does not conflict with any other service.
